aws_signing_helper/update.go

```diff
@@ -0,0 +1,200 @@
+package aws_signing_helper
+
+import (
+	"bufio"
+	"log"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const UpdateRefreshTime = time.Minute * time.Duration(5)
+const AwsSharedCredentialsFileEnvVarName = "AWS_SHARED_CREDENTIALS_FILE"
+
+// Structure to contain a temporary credential
+type TemporaryCredential struct {
+	AccessKeyId     string
+	SecretAccessKey string
+	SessionToken    string
+	Expiration      time.Time
+}
+
+// Updates credentials in the credentials file for the specified profile
+func Update(credentialsOptions CredentialsOpts, profile string, once bool) {
+	var refreshableCred = TemporaryCredential{}
+	var nextRefreshTime time.Time
+	for {
+		credentialProcessOutput, err := GenerateCredentials(&credentialsOptions)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Assign credential values
+		refreshableCred.AccessKeyId = credentialProcessOutput.AccessKeyId
+		refreshableCred.SecretAccessKey = credentialProcessOutput.SecretAccessKey
+		refreshableCred.SessionToken = credentialProcessOutput.SessionToken // nosemgrep
+		refreshableCred.Expiration, _ = time.Parse(time.RFC3339, credentialProcessOutput.Expiration)
+		if (refreshableCred == TemporaryCredential{}) {
+			log.Println("no credentials created")
+			syscall.Exit(1)
+		}
+		// Get/create the credentials file path
+		readOnlyCredentialsFile, err := GetOrCreateReadOnlyCredentialsFile()
```

Comment:
If `readOnlyCredentialsFile` isn't really manipulated, and all we care about are the lines, why not replace `GetOrCreateReadOnlyCredentialsFile()` just be `GetCredentialsFileContents()` and return `[]string` (empty if non-existent)

Reply:
I didn't exactly do this - if there is an error in the middle of the function that is unexpected, I chose to return the error instead. The calling function can check whether an error is returned and handle it appropriately. I did do the other refactor that you suggested in the next revision though.